tests/unit_tests/cuda/detector_cuda_kernel.cu

```diff
@@ -105,4 +105,42 @@ void detector_test(
     DETRAY_CUDA_ERROR_CHECK(cudaDeviceSynchronize());
 }
 
+// cuda kernel to test enumeration
+__global__ void enumerate_test_kernel(
+    detector_view<detector_host_t> det_data,
+    vecmem::data::jagged_vector_view<surface_t> surfaces_data) {
+
+    // Get detector in device
+    detector_device_t detector(det_data);
+
+    // Get surface vector per volume
+    vecmem::device_vector<surface_t> surfaces_device(
+        surfaces_data.m_ptr[threadIdx.x]);
+
+    // Get volume
+    auto& vol = detector.volume_by_index(threadIdx.x);
+
+    // Push_back surfaces to the surface vector
+    for (const auto [obj_idx, obj] : enumerate(detector.surfaces(), vol)) {
+        surfaces_device.push_back(obj);
+    }
+}
+
+// implementation of a test function for surface enumeration
+void enumerate_test(detector_view<detector_host_t> det_data,
+                    vecmem::data::jagged_vector_view<surface_t> surfaces_data) {
+
+    constexpr int block_dim = 1;
+
+    // number of threads = number of volumes
+    int thread_dim = surfaces_data.m_size;
```

Comment:
I know the test is set up such that this would work, but you may still want to check here that this is a "reasonably small" value.

Reply:
I changed it into the general format.